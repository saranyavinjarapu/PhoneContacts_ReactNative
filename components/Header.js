import React from "react";
import { StyleSheet, View, Text } from "react-native";

const Header = (props) => {
  return (
    <View>
      <Text>{props.title}</Text>
    </View>
  );
};

const styles = StyleSheet.create({});

export default Header;
