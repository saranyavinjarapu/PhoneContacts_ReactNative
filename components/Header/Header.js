import React from "react";
import { View, Text } from "react-native";
import styles from "./HeaderStyles";

const Header = (props) => {
  return (
    <View style={styles.header}>
      <Text style={styles.headerTitle} testID={"header_text"}>
        {props.title}
      </Text>
    </View>
  );
};

export default Header;
