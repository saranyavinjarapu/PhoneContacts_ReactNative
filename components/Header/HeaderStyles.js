import { StyleSheet } from "react-native";
import Colors from "../../constants/Colors";

const styles = StyleSheet.create({
  header: {
    width: "100%",
    height: 130,
    paddingTop: 36,
    backgroundColor: Colors.primary,
    alignItems: "center",
    justifyContent: "center",
  },
  headerTitle: {
    color: "black",
    fontSize: 30,
  },
});

export default styles;
