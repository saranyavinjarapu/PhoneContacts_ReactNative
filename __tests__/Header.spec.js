import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Header from "../components/Header/Header";

describe("Header Renders Correctly", () => {
  it("renders correctly", () => {
    const snap = render(<Header />).toJSON();
    expect(snap).toMatchSnapshot();
  });
  it("Header title Works", () => {
    const headerRender = render(<Header title="Hello Test!" />);
    expect(headerRender).toBeTruthy();
    const headerText = headerRender.getByTestId("header_text");
    expect(headerText.props.children).toEqual("Hello Test!");
  });
});
