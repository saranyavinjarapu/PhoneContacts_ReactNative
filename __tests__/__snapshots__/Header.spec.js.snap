// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders Correctly renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f7287b",
      "height": 130,
      "justifyContent": "center",
      "paddingTop": 36,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 30,
      }
    }
    testID="header_text"
  />
</View>
`;
